sycl/source/detail/usm/usm_impl.cpp

```diff
@@ -230,5 +234,60 @@ void *aligned_alloc(size_t Alignment, size_t Size, const queue &Q, alloc Kind) {
   return aligned_alloc(Alignment, Size, Q.get_device(), Q.get_context(), Kind);
 }
 
+// get_pointer_info
+std::tuple<alloc, device> get_pointer_info(const void *Ptr,
+                                           const context &Ctxt) {
+  if (Ctxt.is_host()) {
+    return std::make_tuple(alloc::host, Ctxt.get_devices()[0]);
+  }
+  std::shared_ptr<detail::context_impl> CtxImpl = detail::getSyclObjImpl(Ctxt);
+  std::shared_ptr<detail::usm::USMDispatcher> Dispatch =
+      CtxImpl->getUSMDispatch();
+  pi_context C = CtxImpl->getHandleRef();
+  cl_unified_shared_memory_type_intel AllocTy;
+  pi_device DeviceId;
+
+  // All these CL enums should be replicated in PI
+  detail::pi::checkPiResult(
+      Dispatch->getMemAllocInfo(C, Ptr, CL_MEM_ALLOC_TYPE_INTEL,
+                                sizeof(cl_mem_info_intel), &AllocTy, nullptr));
+  detail::pi::checkPiResult(
+      Dispatch->getMemAllocInfo(C, Ptr, CL_MEM_ALLOC_DEVICE_INTEL,
+                                sizeof(pi_device), &DeviceId, nullptr));
+
+  alloc ResultAlloc;
+  if (AllocTy == CL_MEM_TYPE_HOST_INTEL) {
+    ResultAlloc = alloc::host;
+  } else if (AllocTy == CL_MEM_TYPE_DEVICE_INTEL) {
+    ResultAlloc = alloc::device;
+  } else if (AllocTy == CL_MEM_TYPE_SHARED_INTEL) {
+    ResultAlloc = alloc::shared;
+  } else {
+    ResultAlloc = alloc::unknown;
+  }
+
+  // check device id
+  device Result;
+  if (DeviceId == nullptr) {
+    // Host allocs don't return a device id.
+    // It's irrelevant, so don't do anything in this case.
```

Comment:
Shouldn't we return a host device in this case?

Reply:
I actually did that first, but device(host_selector{}) doesn't work when you run the program with env SYCL_DEVICE_TYPE=CPU (which feels like a bug).

Reply:
I agree. It's a bug.
I would like to avoid piling another "bugs" on top of that.
If we leave it as is, someone will have to remember to update this code + add a host device check to the test when device_selector is fixed. Let's reduce the problem to the test issue only.

To avoid issues with testing we can:
1. Temporarily disable new test on OpenCL CPU and add a TODO comment to enable it back when device selector bug is fixed.
2. To avoid disabling all test cases on OpenCL CPU, we can disable only "host" cases with `#ifdef`, so there will be another build+run or separate test for that.

Thoughts?

Reply:
Is your concern here that the returned device is valid?  The semantics when the return is a host allocation should say to ignore the device in this case.